pandas/core/groupby.py

```diff
@@ -2776,16 +2776,11 @@ def is_in_obj(gpr):
 
         elif is_in_axis(gpr):  # df.groupby('name')
             if gpr in obj:
-                if validate and gpr in obj.index.names:
-                    warnings.warn(
-                        ("'%s' is both a column name and an index level.\n"
-                         "Defaulting to column but "
-                         "this will raise an ambiguity error in a "
-                         "future version") % gpr,
-                        FutureWarning, stacklevel=5)
+                if validate:
+                    obj._check_column_or_level_ambiguity(gpr)
                 in_axis, name, gpr = True, gpr, obj[gpr]
                 exclusions.append(name)
-            elif gpr in obj.index.names:
+            elif obj._is_index_reference(gpr):
```

Comment:
this should fail on a Series the way it works currently

Reply:
Thanks for pointing this out @jreback.  I'm still working on a full set of updates for this PR but in the meantime, I added a test case to #17843 that would have caught this.